dms/models/directory.py

```diff
@@ -475,6 +475,14 @@ def _compute_directory_type(self):
                 record.parent_id = None
             else:
                 record.root_storage_id = None
+                if self.env.context.get(
+                    "active_model"
+                ) == self._name and self.env.context.get("active_id"):
+                    record.parent_id = self.env.context.get("active_id")
+                else:
+                    record.parent_id = self.env["dms.directory"].search(
```

Comment:
Searching for the first directory is OK, being from any storage?

Reply:
IMO yes because not set yet storage (storage auto-defined when `parent_id` is set).
According to create directories, exists 3 possibilities:
- Create manual directory from storage smartbutton called "_Directories_": In these case auto-set `is_root_directory` to True and `parent_id` is not necessary.
- Create manual directory from other directory smartbutton called "_Subdirectories_": In these case `active_id` auto-set these record to `parent_id`.
- Create manual directory: Auto-set first root directory search from any storage.

Reply:
The name of the compute should now be changed, isn't it?